src/Agent.Worker/Container/ContainerInfo.cs

```diff
@@ -135,6 +182,78 @@ public string TranslateToHostPath(string path)
 
             return path;
         }
+
+        private List<MountVolume> ParseVolumes(IList<string> volumes)
+        {
+            List<MountVolume> mountVolumes = new List<MountVolume>();
+            if (volumes?.Count > 0)
+            {
+                foreach (var volume in volumes)
+                {
+                    var volumeSplit = volume.Split(":");
+                    if (volumeSplit?.Length == 3)
+                    {
+                        // source:target:ro
+                        mountVolumes.Add(new MountVolume(volumeSplit[0], volumeSplit[1], String.Equals(volumeSplit[2], "ro")));
+                    }
+                    else if (volumeSplit?.Length == 2)
+                    {
+                        if (String.Equals(volumeSplit[1], "ro"))
+                        {
+                            // target:ro
+                            mountVolumes.Add(new MountVolume(null, volumeSplit[0], true));
+                        }
+                        else
+                        {
+                            // source:target
+                            mountVolumes.Add(new MountVolume(volumeSplit[0], volumeSplit[1]));
+                        }
+                    }
+                    else
+                    {
+                        // target - or, default to passing straight through
+                        mountVolumes.Add(new MountVolume(null, volume));
```

Comment:
should we error and ignore if this is not a valid setting?

Reply:
I chose to just pass the full string though so if a bad option is given by the user in yaml, it will pass all the way through to the `docker run -p` or `docker run -v` so we can leverage the existing docker error messages for bad inputs here